### PR TITLE
Update project configuration and sln files

### PR DESCRIPTION
**Pull request by karajas**
/cc @weshaggard 

Reimplementation of https://github.com/dotnet/corefx/pull/14872

### CODE REVIEW

File: src/Common/tests/Common.Tests.csproj [hunk 1]
```diff
@@ -7,14 +7,10 @@
...
   <!-- Default configurations to help VS understand the configurations -->
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'FreeBSD_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'FreeBSD_Release|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Linux_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Linux_Release|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'OSX_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'OSX_Release|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Release|AnyCPU'" />
+  <PropertyGroup Condition = "'$(Configuration)|$(Platform)' == 'netstandard1.3-Unix-Debug|AnyCPU'" />
```
Comment (ellismg) [inline src/Common/tests/Common.Tests.csproj]: NIT: Is it possible to not have spaces between the = after the condition, eg:

```diff
-  <PropertyGroup Condition = "'$(Configuration)|$(Platform)' == 'netstandard1.3-Unix-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard1.3-Unix-Debug|AnyCPU'" />
```

Comment (weshaggard) [inline src/Common/tests/Common.Tests.csproj]: Yes please lets collapse the extra spaces.

**Comment by weshaggard** [comment]
You will need to merge with https://github.com/dotnet/corefx/pull/15181. Lets please get these regenerated and in today.

**Comment by karajas** [comment]
Updated.

**Comment by karajas** [comment]
For reference issues with VS sln builds, deleting the generated .vs hidden directory adjacent to the sln file should help resolve some issues. Also be sure not to have any *.user files in the src and test folders.

File: src/Common/tests/Common.Tests.csproj [hunk 2]
```diff
@@ -7,14 +7,10 @@
...
   <!-- Default configurations to help VS understand the configurations -->
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'FreeBSD_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'FreeBSD_Release|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Linux_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Linux_Release|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'OSX_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'OSX_Release|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='netstandard1.3-Unix-Debug|Any CPU'" />
```
Comment (weshaggard) [inline src/Common/tests/Common.Tests.csproj]: `Any CPU` should be `AnyCPU` without a space.

**Comment by weshaggard** [comment]
Other then the space in AnyCPU the change looks good from a quick glance.

How easy is it to regenerate these? I ask because we are going to be tweaking a lot of configurations and it would be good to have a way to generate these.

**Comment by karajas** [comment]
Will resolve the Any CPU change. Think I regressed when tracking down a separate issue where it wasn't loading the right configs.

Right now it's just a console app I run with a set of rules. If we want to ensure that they stay in sync like the csproj, I can add another opt-in regen/validate step to tools.

**Comment by weshaggard** [comment]
Yes please add that to the custom task which folks can run to fix up the configurations.

**Comment by karajas** [comment]
We have one for csprojs in this change, do we need one for sln files?

**Comment by weshaggard** [comment]
Yes I would include the on for sln files as well.

**Comment by karajas** [comment]
I'll add it as a task after getting this merged to unblock some people.

**Comment by karajas** [comment]
For reference, the configs in VS are currently in the format (Debug/Release)(TestPlatform)(SrcProjectConfiguration)(TestProjectConfiguration). The idea was the TestPlatform would determine the test runner. Currently we only support TestPlatform = NETCoreApp.

@mellinoe pointed out that some of the configurations don't make sense per se. For example, building src for netstandard and test for netstandard then trying to run for netcoreapp. Currently the idea was to allow people to build for the config they want after making changes to see if that passes. I can look into cleaning this up if required.

**Comment by weshaggard** [comment]
>@mellinoe pointed out that some of the configurations don't make sense per se. For example, building src for netstandard and test for netstandard then trying to run for netcoreapp. Currently the idea was to allow people to build for the config they want after making changes to see if that passes. I can look into cleaning this up if required.

Why doesn't it make sense to build as netstandard and run as netcoreapp? That is a valid scenario and interesting for a lot of our projects.

**Comment by mellinoe** [comment]
I think he meant "build as net63, and run as netcoreapp", or things along those lines. There were some combinations that didn't make sense to me.

**Comment by karajas** [comment]
Sorry you're right that is a valid config. I took the cross product of src and test configs, and things like building src for net46 and building tests for netcoreapp is a better example. Or maybe building on Windows and trying to run the Unix bits for example.

**Comment by weshaggard** [comment]
Yes building for net463 and running on netcoreapp doesn't many any sense. There are also OS configurations that also don't make sense as you would run the OSX build on netcoreapp on windows. While the F5 scenario doesn't make sense it still make sense to be able to build those in VS so I think it is OK to keep those.

**Comment by weshaggard** [comment]
So I think we should keep those build configurations and at best we should add some checks to give nice error messages if you try to run one of those configurations on Windows netcoreapp.

**Comment by KirillOsenkov** [comment]
Folks, this commit is a big problem:
https://github.com/dotnet/corefx/commit/f6ab339f446e37505f93ef9f430801fa1dcc05b8

My understanding is that @karajas has a tool that writes the .sln files. Unfortunately it seems to have a bug where the project nesting section is written out incorrectly. I've seen an anomaly such as this:

https://github.com/dotnet/corefx/commit/f6ab339f446e37505f93ef9f430801fa1dcc05b8#diff-3a4d4461131dc911f789ebf61bf592b7R8

or even this:
https://github.com/dotnet/corefx/commit/f6ab339f446e37505f93ef9f430801fa1dcc05b8#diff-5c06d4b0c4ec0ab4a1c79737881b8c08R8

Building any such .sln file with MSBuild will now result in MSBuild crashing (I've filed a bug https://github.com/Microsoft/msbuild/issues/1599 to make MSBuild more resilient against this situation).

My recommendation would be to patch the tool to correctly write out the ProjectDependencies and NestedProjects section ASAP.

**Comment by KirillOsenkov** [comment]
Opened https://github.com/dotnet/corefx/issues/15378

**Comment by karajas** [comment]
Noticed this while refactoring the tool into a task. @KirillOsenkov I'll make sure to correct this.

**Pull request merged**